targets/TARGET_STM/TARGET_STM32G4/analogin_device.c

```diff
@@ -163,7 +199,7 @@ uint16_t adc_read(analogin_t *obj)
 
     // Configure ADC channel
     sConfig.Rank         = ADC_REGULAR_RANK_1;
-    sConfig.SamplingTime = ADC_SAMPLETIME_247CYCLES_5;
+    sConfig.SamplingTime = ADC_SAMPLETIME_24CYCLES_5;
```

Comment:
Maybe you should keep 247 cycles value,
some tests are failing with 24 value...

Reply:
Perhaps I should increase the ADC clock prescaler?